app/server/components/repositories/teams.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+var permUtil = require('./util/permissions'),
+    repoUtil = require('./util/repos'),
+    teamUtil = require('./util/teams'),
+    getPrefix,
+    getTeam;
+
+getPrefix = (repo) => {
+    return 'zzz-permissive-repo-' + repo.name + '-';
+};
+
+getTeam = (teams, prefix, permission) => {
+    let name = prefix + permission;
+    return teams.find(team => team.name === name);
+};
+
+module.exports = {
+
+    edit (username, repoId, team) {
+
+        // get permission map
+        return permUtil.getPermissionMap().then(map => {
+
+            let users = map[repoId],
+                current = users[username];
+
+            // user already on the team
+            if (current.permissive === team) {
+                return new Error('User already assigned the desired permission');
```

Comment:
I'm not sure I'd throw an error here. If they're already on the team, it could just be silently ignored.

From an HTTP perspective, I'd think of it like an idempotent operation (which PUT already is) - you can call that function over and over, and if the values are the same, nothing happens. With an error, the second call actually has a different result.